protocols/smtp/src/main/java/org/apache/james/protocols/smtp/hook/HookResult.java

```diff
@@ -31,113 +35,81 @@
     private static final HookResult DENYSOFT = new HookResult(HookReturnCode.DENYSOFT);
     private static final HookResult DISCONNECT = new HookResult(HookReturnCode.DISCONNECT);
 
+    public static HookResult declined() {
+        return DECLINED;
+    }
```

Comment:
why not simply making public DECLINED, OK, etc.?

Reply:
I prefer Matthieu connected / diconnected factory proposal, as it convey the information about keeping the connection.